src/pages/Profile.js

```diff
@@ -1,22 +1,59 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, {useContext, useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+import axios from "axios";
+import {AuthContext} from "../context/AuthContext";
 
 function Profile() {
-  return (
-    <>
-      <h1>Profielpagina</h1>
-      <section>
-        <h2>Gegevens</h2>
-        <p><strong>Gebruikersnaam:</strong> hardcoded-test</p>
-        <p><strong>Email:</strong> hardcoded@test.com</p>
-      </section>
-      <section>
-        <h2>Strikt geheime profiel-content</h2>
-        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-      </section>
-      <p>Terug naar de <Link to="/">Homepagina</Link></p>
-    </>
-  );
+
+    const {isAuth, user} = useContext(AuthContext);
+    const token = localStorage.getItem('token');
+    const [privateContent, setPrivateContent] = useState({});
```

Comment:
Ik zou zelf de default waarde aanpassen naar undefined. Dan heb je een beter verschil tussen of het al wel/niet is ingeladen.

Wanneer de getPrivateContent functie niet lukt, dan krijg je errors omdat die `title` en `content` niet kan vinden op het object. Je kunt dan met `undefined` van te voren checken of het al ingeladen is of niet
```suggestion
    const [privateContent, setPrivateContent] = useState();
```